extensions/wikia/EditPageLayout/js/editor/AceEditor.js

```diff
@@ -0,0 +1,199 @@
+define('wikia.editpage.ace.editor', ['wikia.ace.editor'], function(ace){
+	'use strict';
+
+	var theme = 'solarized_light',
+		isEditorWide = false,
+		inputAttr = {
+			name: 'wpTextbox1',
+			id: 'wpTextbox1'
+		},
+		$editPage = $('#EditPage'),
+		$editPageToolbar = $('#EditPageToolbar'),
+		wideClassName = 'editpage-sourcewidemode-on',
+		narrowClassName = 'editpage-sourcewidemode-off',
+		beforeEditorInit = function(){
+			$('.loading-indicator').remove();
+			$('#wpSave').removeAttr( 'disabled' );
+			$editPage.addClass('editpage-sourcewidemode mode-source ' + narrowClassName);
+		};
+
+	function init() {
+		chooseTheme();
+		initConfig();
+		beforeEditorInit();
+		ace.init( 'editarea', inputAttr );
+		initOptions();
+		initStyles();
+		ace.setTheme( theme );
+		ace.setMode( window.codePageType );
+
+		initEvents();
+	}
+
+	/**
+	 * Init ace editor config
+	 */
+	function initConfig() {
+		var config = {
+				workerPath: window.aceScriptsPath
+			};
+
+		ace.setConfig( config );
+	}
+
+	/**
+	 * Init ace editor options
+	 */
+	function initOptions() {
+		var options = {
+			showPrintMargin: false
+		};
+
+		ace.setOptions( options );
+	}
+
+	/**
+	 * Init ace editor custom styles
+	 */
+	function initStyles() {
+		var editor = ace.getEditor();
+
+		editor.css('fontSize', '14px');
+	}
+
+	/**
+	 * Init submit action
+	 */
+	function initSubmit() {
+		$( '#editform' ).submit(function(e) {
+			var $form = $( this ),
+				hiddenInput = ace.getInput().val( ace.getContent() );
+
+			$form.append( hiddenInput );
+			$form.submit();
+
+			e.preventDefault();
+		});
+	}
+
+	/**
+	 * Choose ace editor theme depending on wiki theme (light or dark)
+	 */
+	function chooseTheme() {
+		if (window.wgIsDarkTheme) {
```

Comment:
Since code is making use of window, there's an AMD module 'wikia.window' that could be passed as a dependency here :)